Fix division by zero as stmmac clock is not yet enabled.

This fixes the following warning during boot:
[<c0110b80>] (unwind_backtrace) from [<c010ad24>] (show_stack+0x18/0x1c)
[<c010ad24>] (show_stack) from [<c09cf23c>] (dump_stack_lvl+0x40/0x4c)
[<c09cf23c>] (dump_stack_lvl) from [<c05ffe34>] (Ldiv0+0x8/0x10)
[<c05ffe34>] (Ldiv0) from [<c070a694>] (config_sub_second_increment+0x30/0x78)
[<c070a694>] (config_sub_second_increment) from [<c06fd168>] (stmmac_init_tstamp_counter+0x108/0x1bc)
[<c06fd168>] (stmmac_init_tstamp_counter) from [<c07032b8>] (stmmac_hw_setup+0x73c/0xe84)
[<c07032b8>] (stmmac_hw_setup) from [<c0703ef0>] (stmmac_open+0x4f0/0xb8c)
[<c0703ef0>] (stmmac_open) from [<c082a284>] (__dev_open+0x138/0x184)
[<c082a284>] (__dev_open) from [<c082a604>] (__dev_change_flags+0x190/0x1b0)
[<c082a604>] (__dev_change_flags) from [<c082a640>] (dev_change_flags+0x1c/0x4c)
[<c082a640>] (dev_change_flags) from [<c08d9e24>] (devinet_ioctl+0x2d4/0x5dc)
[<c08d9e24>] (devinet_ioctl) from [<c08dcd7c>] (inet_ioctl+0x208/0x238)
[<c08dcd7c>] (inet_ioctl) from [<c0804850>] (sock_ioctl+0x164/0x3ec)
[<c0804850>] (sock_ioctl) from [<c02e0860>] (vfs_ioctl+0x28/0x3c)
[<c02e0860>] (vfs_ioctl) from [<c02e1240>] (sys_ioctl+0xe4/0x944)
[<c02e1240>] (sys_ioctl) from [<c0100060>] (ret_fast_syscall+0x0/0x48)
Exception stack(0xc1375fa8 to 0xc1375ff0)

diff -aurN a/drivers/net/ethernet/stmicro/stmmac/stmmac_platform.c c/drivers/net/ethernet/stmicro/stmmac/stmmac_platform.c
--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_platform.c	2022-03-28 08:11:23.941356152 +0200
+++ c/drivers/net/ethernet/stmicro/stmmac/stmmac_platform.c	2022-03-28 10:32:40.436106056 +0200
@@ -579,7 +579,7 @@
 	}
 
 	/* clock setup */
-	if (!of_device_is_compatible(np, "snps,dwc-qos-ethernet-4.10")) {
+	if (IS_ENABLED(CONFIG_ARCH_NETX4000) || !of_device_is_compatible(np, "snps,dwc-qos-ethernet-4.10")) {
 		plat->stmmac_clk = devm_clk_get(&pdev->dev,
 						STMMAC_RESOURCE_NAME);
 		if (IS_ERR(plat->stmmac_clk)) {
